octree/include/pcl/octree/impl/octree_iterator.hpp

```diff
@@ -281,6 +281,65 @@ namespace pcl
 
       return (*this);
     }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeFixedDepthIterator<OctreeT>::OctreeFixedDepthIterator (unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (depth_arg), depth_(depth_arg)
```

Comment:
I just realized what you meant before with the setting the max_depth on the base constructor. In order to take advantage of that reset method I propose, you must initialized max_depth to 0 always and let it assume the depth of the tree. This means that you'll just invoke the default constructor in this case or explicitly pass 0.

Reply:
Ok, I set `max_depth` to 0. So now the `OctreeFixedDepthIterator` is fully in charge to check its good behavior.

Reply:
In the recent commits, I change my mind and in the `OctreeBreadthFirstIterator::reset()` that I proposed/updated, I set the `max_depth` value to `fixed_depth_arg`. This is especially usefull when `fixed_depth_arg` == 0. I wrote it in the general comment. This really simplies the `OctreeBreadthFirstIterator::operator++()` and it's better in my opinion, it effectively reuse what have been done in the previous classes, so that perfect for me.

So now, I would like to call the base constructor with the `fixed_depth_arg` parameter. Do you agree with that?

Reply:
I made some change regarding this point in the commit 00ce6b209d3e3f9f6a1f720f147e5945ddffcc08

Let me know if you validate it or not. For me, the behavior of the iterator is good this way.